tests/kafkatest/services/replica_verification_tool.py

```diff
@@ -0,0 +1,82 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.services.background_thread import BackgroundThreadService
+
+from kafkatest.services.kafka.directory import kafka_dir
+from kafkatest.services.security.security_config import SecurityConfig
+
+import re
+
+class ReplicaVerificationTool(BackgroundThreadService):
+
+    logs = {
+        "producer_log": {
+            "path": "/mnt/replica_verification_tool.log",
+            "collect_default": False}
+    }
+
+    def __init__(self, context, num_nodes, kafka, topic, report_interval_ms, security_protocol="PLAINTEXT"):
+        super(ReplicaVerificationTool, self).__init__(context, num_nodes)
+
+        self.kafka = kafka
+        self.topic = topic
+        self.report_interval_ms = report_interval_ms
+        self.security_protocol = security_protocol
+        self.security_config = SecurityConfig(security_protocol)
+        self.lagged_values = []
+        self.zero_lagged_values = []
+
+    def _worker(self, idx, node):
+        cmd = self.start_cmd(node)
+        self.logger.debug("ReplicaVerificationTool %d command: %s" % (idx, cmd))
+        self.security_config.setup_node(node)
+        for line in node.account.ssh_capture(cmd):
+            lag = re.search('.*max lag is (.+?) for partition', line)
+            if lag:
+                if int(lag.group(1)) > 0:
+                    self.logger.info("Appending to lags: %s" % line)
+                    self.lagged_values.append(line)
+                else:
+                    self.logger.info("Appending to zero lags: %s" % line)
+                    self.zero_lagged_values.append(line)
+
+    def num_lags(self):
```

Comment:
I'm not sure `num_lags` and `num_zero_lags` are exposing exactly what we want.

It seems like, among other things, the replica verification tool is printing a stream of updates to "max lag" state for different topic-partitions

Rather than essentially keeping count of the number of times a replica has been behind or has been fully caught up (for any topic-partition), would it make sense to keep a map from topic-partition to max lag which gets updated each time you read in a line from the tool?

Also - would it make sense to expose any of the checksum information?